### PR TITLE
chore: update golangci-lint to v1.31

### PR DESCRIPTION
**Pull request by grace0925**
Signed-off-by: Grace Liu <grace.liu@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

File: pkg/internal/cryptosetup/cryptosetup.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ func prepareKeyHandle(storeProvider storage.Provider, keyManager kms.KeyManager,
 	var kh *keyset.Handle
 
 	keyIDBytes, err := keyIDStore.Get(keyIDDBKeyName)
-	if err != nil {
+	if err != nil { //nolint:nestif
```
Comment (rolsonquadras) [inline pkg/internal/cryptosetup/cryptosetup.go:115]: Is there any plan to fix minor lint issues like this in next PRs ?

**Review by rolsonquadras — APPROVED** [review]

File: pkg/proxy/rules/filerules/provider.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ type Provider struct {
 
 // New returns new proxy rules provider from file
 func New(configFile string) (*Provider, error) {
-	data, err := ioutil.ReadFile(configFile) //nolint: gosec
+	data, err := ioutil.ReadFile(configFile) //nolint:gosec
```
Comment (DRK3) [inline pkg/proxy/rules/filerules/provider.go:45]: @fqutishat @rolsonquadras Does this warrant a followup story?

File: test/bdd/dockerutil/compose.go [hunk 3]
```diff
@@ -152,7 +152,7 @@ func (c *Composition) GenerateLogs(dir, logName string) error {
 		return err
 	}
 
-	f, err := os.OpenFile(logName, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0600)
+	f, err := os.OpenFile(logName, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0600) //nolint:gosec
```
Comment (DRK3) [inline test/bdd/dockerutil/compose.go:155]: @fqutishat @rolsonquadras Does this warrant a followup story?

(we should be careful about disable the security check)

File: pkg/proxy/rules/filerules/provider.go (hunk 2 above)
Comment (DRK3) [reply pkg/proxy/rules/filerules/provider.go:45]: Just realized this line wasn't really changed

File: pkg/internal/cryptosetup/cryptosetup.go (hunk 1 above)
Comment (grace0925) [reply pkg/internal/cryptosetup/cryptosetup.go:115]: I can create an issue and refactor these minor ones

**Review by sudeshrshetty — APPROVED** [review]

File: test/bdd/dockerutil/compose.go (hunk 3 above)
Comment (fqutishat) [reply test/bdd/dockerutil/compose.go:155]: Bdd test not important

Comment (fqutishat) [reply test/bdd/dockerutil/compose.go:155]: Your comment is correct if it's part of the code

File: pkg/proxy/rules/filerules/provider.go (hunk 2 above)
Comment (grace0925) [reply pkg/proxy/rules/filerules/provider.go:45]: I'm not sure why but linter complains about the space

File: test/bdd/dockerutil/compose.go (hunk 3 above)
Comment (DRK3) [reply test/bdd/dockerutil/compose.go:155]: Ah, makes sense. I didn't realize it was in the bdd test.

**Pull request merged**